esp8266-fastled-webserver/Fields.cpp

```diff
@@ -17,479 +17,526 @@
 */
 #include "common.h"
 
-const String NumberFieldType = "Number";
-const String BooleanFieldType = "Boolean";
-const String SelectFieldType = "Select";
-const String ColorFieldType = "Color";
-const String SectionFieldType = "Section";
-const String StringFieldType = "String";
-const String LabelFieldType = "Label";
-const String UtcOffsetIndexFieldType = "UtcOffsetIndex";
-
-uint8_t power = 1;
-uint8_t brightness = brightnessMap[brightnessIndex];
-
-//String setPower(String value) {
-//  power = value.toInt();
-//  if(power < 0) power = 0;
-//  else if (power > 1) power = 1;
-//  return String(power);
-//}
-
-String getPower() {
-  return String(power);
-}
-
-//String setBrightness(String value) {
-//  brightness = value.toInt();
-//  if(brightness < 0) brightness = 0;
-//  else if (brightness > 255) brightness = 255;
-//  return String(brightness);
-//}
-
-// TODO: wrapper class for EEPROM settings, to help ensure clear understanding
-//       of all locations that can cause settings committed / EEPROM write.
-//       Also simplifies reading prior value before writing new value (extend EEPROM life)
-//       Also simplifies later extending CRC, transactional updates, bypassing "dead"
-//       areas of EEPROM, etc.
-
-String getBrightness() {
-  return String(brightness);
-}
-
-String getPattern() {
-  return String(currentPatternIndex);
-}
-
-String getPatterns() {
-  String json = "";
+// only items outside the anonymous namespace can be exported.
+// this keeps the global namespace cleaner, with no runtime/RAM costs.
+inline namespace {
+  enum struct Field_t : uint8_t {
+    Number,
+    Boolean,
+    Select,
+    Color,
+    Section,
+    String,
+    Label,
+    UtcOffset,
+  };
```

Comment:
```suggestion
    Label,
    UtcOffset,
    
    Invalid,
  };
```

Reply:
While some designs do include an explicit "invalid" value for an enumeration, there are definitely reasons to avoid this (at least with scoped enumerations).

One benefit of scoped enumerations (`enum struct ...` or `enum class`) is that, when writing a `switch` statement, the compiler will generate a warning when there is a value that is not handled by a `case` statement.

In fact, when adding the UtcOffset, I avoided a bug by this exact warning.

If adding an explicit 'invalid' value, then it might suggest that value is the only invalid value (which is not the case).  Moreover, for this type, there should *never* be an invalid value.

Again, I realize this is a common pattern.  I did consider it, but given the enum's purposes and use model, I explicitly chose not to add it here.